### PR TITLE
recurse-dir: add new readdir_all_by_path() helper

### PR DESCRIPTION
**Pull request by poettering**
This new helper combines open() with readdir_all() to simplify a few callers.

### CODE REVIEW

**Comment by poettering** [comment]
just to shorten the code. doesn't change behaviour, and doesn't fix anything. hence let's play safe and leave it for v256

**Review by yuwata — APPROVED** [review]

**Comment by yuwata** [comment]
CentOS CI (Arch + sanitizers) failure is #30056, unrelated.

File: src/basic/recurse-dir.c [hunk 1]
```diff
@@ -132,6 +133,18 @@ int readdir_all(int dir_fd,
         return 0;
 }
 
+int readdir_all_by_path(int fd, const char *path, RecurseDirFlags flags, DirectoryEntries **ret) {
```
Comment (daandemeyer) [inline src/basic/recurse-dir.c]: Can we name this `readdir_all_at()`? I very much prefer that we use the same naming scheme for all these functions that work in an openat() style.

**Review by DaanDeMeyer — CHANGES_REQUESTED** [review]

File: src/basic/recurse-dir.c (hunk 1 above)
Comment (poettering) [reply src/basic/recurse-dir.c]: addressed

**Pull request merged**